src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as todoActions } from '../../features/currentTodo';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const dispatch = useAppDispatch();
+  const { currentTodo } = useAppSelector(state => state);
```

Comment:
grab only what you really need
```suggestion
  const currentTodo = useAppSelector(state => state.currentTodo ;
```

Reply:
fix throughout your code